docs/checkit.js

```diff
@@ -907,6 +981,16 @@ CheckIt.prototype.addCalendarToDropdown = function (uniqueId, title, $dropdown)
     //add the calendar with unique Id, uniqueId, and title, title, to
     //the saved calendars dropdown on the navbar.
     "use strict";
+    
+    // first check if the title isn't already in the dropdown
+    if ($('#' + uniqueId).length) {
+        // title is already in dropdown
+        // but the title may have changed
+        // so just update the title
+        $('#' + uniqueId).find("a").html(title);
```

Comment:
Probably want to sanitize `title` and not just stuff it in as HTML. Does `innerText` or something do it? I don't remember. Anyway, you never want to let the browser evaluate unsanitized user input, be it JS, HTML, CSS, whatever.

Reply:
What do you mean sanitize?

Reply:
In general, sanitization is the process of taking untrusted data (like from a user) and attempting to make it safe to use as an input to something else (like displaying on a web page. It's hard to do correctly most of the time. It's the reason you write e.g. SQL queries using `?` in place of variables and let the sql engine itself do the substitution. If you were to just build the whole query string yourself but I'm allowed to add arbitrary text to the middle of it, I can write something clever that will screw everything up.

In this case, you're taking the title of the calendar, which the user has total control over, and stuffing it into an HTML node in the DOM, which means the browser is going to evaluate it. I could make my title be something crazy like a `<script>` tag that contains javascript that uploads random garbage data until your app runs out of space or you get a giant bill from Google for resource usage, etc.

I think what you want here is [createTextNode](https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode).

Maybe something like this (untested)?

```js
var $elem = $('#' + uniqueId).find('a');
$elem.empty()
$elem.createTextNode(title);
```

Reply:
I opened up an issue for this.